lib/membrane_element_portaudio/sink.ex

```diff
@@ -45,62 +46,55 @@ defmodule Membrane.Element.PortAudio.Sink do
   @impl true
   def handle_init(%__MODULE__{} = options) do
     {:ok,
-     %{
-       endpoint_id: options.endpoint_id,
-       ringbuffer_size: options.ringbuffer_size,
-       pa_buffer_size: options.portaudio_buffer_size,
-       latency: options.latency,
-       native: nil,
-       playing: false
-     }}
+     options
+     |> Map.from_struct()
+     |> Map.merge(%{
+       native: nil
+     })}
   end
 
   @impl true
-  def handle_play(state) do
+  def handle_prepared_to_playing(_ctx, state) do
     %{
       endpoint_id: endpoint_id,
       ringbuffer_size: ringbuffer_size,
-      pa_buffer_size: pa_buffer_size,
+      portaudio_buffer_size: pa_buffer_size,
       latency: latency
     } = state
 
-    state = %{state | playing: true}
-
     endpoint_id = if endpoint_id == :default, do: @pa_no_device, else: endpoint_id
 
     with {:ok, native} <-
-           @native.create_sink(self(), endpoint_id, ringbuffer_size, pa_buffer_size, latency) do
+           SyncExecutor.apply(Native, :create, [
+             self(),
+             endpoint_id,
+             ringbuffer_size,
+             pa_buffer_size,
+             latency
+           ]) do
       {:ok, %{state | native: native}}
     else
       {:error, reason} -> {{:error, reason}, state}
     end
   end
 
   @impl true
-  def handle_prepare(:playing, %{native: native} = state) do
-    {@native.destroy_sink(native), %{state | native: nil, playing: false}}
-  end
-
-  @impl true
-  def handle_prepare(_, state) do
-    {:ok, state}
+  def handle_playing_to_prepared(_ctx, %{native: native} = state) do
+    {SyncExecutor.apply(Native, :destroy, native), %{state | native: nil}}
   end
 
   @impl true
-  def handle_other(
-        {:membrane_element_portaudio_ringbuffer_demand, size},
-        %{playing: true} = state
-      ) do
-    {{:ok, demand: {:sink, size}}, state}
+  def handle_other({:demand, size}, %{playback_state: :playing}, state) do
+    {{:ok, demand: {:input, &(&1 + size)}}, state}
```

Comment:
In `sink.c`(L31) you have already changed demand calculation to include the demand. Here you increase the demand again. Is that an error or am I missing something?